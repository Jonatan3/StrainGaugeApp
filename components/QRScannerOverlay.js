import React from 'react'
import { View, Text, TouchableWithoutFeedback, Dimensions } from 'react-native'
import { RNCamera } from 'react-native-camera'
import { Arrow, Flash_off, Flash_on } from '../resources/svgs'
import colors from '../resources/colors'

export default function QRScannerOverlay({ onFlashClick, flash, navigation }) {

  const SCREEN_WIDTH = Dimensions.get("window").width;

  return (
    <View style={{ flex: 1, width: SCREEN_WIDTH }}>
      {/* Top part of the greyed out area, and contain buttons and text in the top */}
      <View style={{ flex: 1, flexDirection: 'row', backgroundColor: colors.cameraNoFocus }}>
        <View style={{ marginTop: 24, flex: 1, justifyContent: 'space-around', flexDirection: 'row' }} >
          <TouchableWithoutFeedback onPress={() => navigation.pop()} >
            <View>
              <Arrow style={{ justifyContent: 'space-around' }} />
            </View>
          </TouchableWithoutFeedback>
        </View>
        <View style={{ flex: 4, justifyContent: 'space-around', flexDirection: 'row' }}>
          <Text style={{ marginTop: 24, color: 'white', fontSize: 16 }}>
            Scan QR to navigate to channel
          </Text>
        </View>
        <View style={{ marginTop: 24, flex: 1, justifyContent: 'space-around', flexDirection: 'row' }}>
          <TouchableWithoutFeedback onPress={() => onFlashClick()} >
            <View>
              {flash === RNCamera.Constants.FlashMode.off ? <Flash_off /> : <Flash_on />}
            </View>
          </TouchableWithoutFeedback>
        </View>
      </View>
      {/* Middel part of the overlay */}
      <View style={{ flex: 1, flexDirection: 'row' }}>
        <View style={{ flex: 1, backgroundColor: colors.cameraNoFocus }} />

        <View style={{ flex: 3, flexDirection: 'row', backgroundColor: 'transparent' }} >
          <View style={{ flex: 1, flexDirection: 'column' }}>
            <View style={{ flex: 1, borderTopWidth: 2, borderLeftWidth: 2, borderColor: 'white', marginTop: 1, marginLeft: 1 }} />
            <View style={{ flex: 4 }} />
            <View style={{ flex: 1, borderBottomWidth: 2, borderLeftWidth: 2, borderColor: 'white', marginBottom: 1, marginLeft: 1 }} />
          </View>
          <View style={{ flex: 4 }} />
          <View style={{ flex: 1, flexDirection: 'column' }}>
            <View style={{ flex: 1, borderTopWidth: 2, borderRightWidth: 2, borderColor: 'white', marginTop: 1, marginRight: 1 }} />
            <View style={{ flex: 4 }} />
            <View style={{ flex: 1, borderBottomWidth: 2, borderRightWidth: 2, borderColor: 'white', marginBottom: 1, marginRight: 1 }} />
          </View>
        </View>
        <View style={{ flex: 1, backgroundColor: colors.cameraNoFocus }} />

      </View>
      {/* last part of the of the overlay */}
      <View style={{ flex: 1, backgroundColor: colors.cameraNoFocus }} />
    </View>
  )
}