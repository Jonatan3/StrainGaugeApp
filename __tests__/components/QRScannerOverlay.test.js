import React from 'react';
import renderer from 'react-test-renderer';
import QRScannerOverlay from '../../components/QRScannerOverlay'

test('renders correctly', () => {
    const tree = renderer.create(<QRScannerOverlay />).toJSON()
    expect(tree).toMatchSnapshot()
})