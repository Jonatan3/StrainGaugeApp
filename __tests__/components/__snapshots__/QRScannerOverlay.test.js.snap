// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "width": 750,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 24,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="24"
          bbWidth="24"
          fill="none"
          focusable={false}
          height="24"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 24,
                "width": 24,
              },
            ]
          }
          vbHeight={15}
          vbWidth={16}
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M7.44049 1.00195L1.00049 7.00195L7.44049 13.002"
              propList={
                Array [
                  "stroke",
                ]
              }
              stroke={4294967295}
            />
            <RNSVGPath
              d="M1.94049 6.99994H14.9405"
              propList={
                Array [
                  "stroke",
                ]
              }
              stroke={4294967295}
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
    <View
      style={
        Object {
          "flex": 4,
          "flexDirection": "row",
          "justifyContent": "space-around",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "marginTop": 24,
          }
        }
      >
        Scan QR to navigate to channel
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "space-around",
          "marginTop": 24,
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="36px"
          bbWidth="36px"
          fill="#eaeaee"
          focusable={false}
          height="36px"
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": 36,
                "width": 36,
              },
            ]
          }
          vbHeight={24}
          vbWidth={24}
          width="36px"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={4293585646}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M0 0h24v24H0V0z"
              fill={null}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M7 3v9c0 .55.45 1 1 1h2v7.15c0 .51.67.69.93.25l5.19-8.9c.39-.67-.09-1.5-.86-1.5H13l2.49-6.65c.25-.65-.23-1.35-.93-1.35H8c-.55 0-1 .45-1 1z"
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "transparent",
          "flex": 3,
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "white",
              "borderLeftWidth": 2,
              "borderTopWidth": 2,
              "flex": 1,
              "marginLeft": 1,
              "marginTop": 1,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 4,
            }
          }
        />
        <View
          style={
            Object {
              "borderBottomWidth": 2,
              "borderColor": "white",
              "borderLeftWidth": 2,
              "flex": 1,
              "marginBottom": 1,
              "marginLeft": 1,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 4,
          }
        }
      />
      <View
        style={
          Object {
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "borderColor": "white",
              "borderRightWidth": 2,
              "borderTopWidth": 2,
              "flex": 1,
              "marginRight": 1,
              "marginTop": 1,
            }
          }
        />
        <View
          style={
            Object {
              "flex": 4,
            }
          }
        />
        <View
          style={
            Object {
              "borderBottomWidth": 2,
              "borderColor": "white",
              "borderRightWidth": 2,
              "flex": 1,
              "marginBottom": 1,
              "marginRight": 1,
            }
          }
        />
      </View>
    </View>
    <View
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,0.5)",
          "flex": 1,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
      }
    }
  />
</View>
`;
